frontend/src/pages/videos/VideoRecommendation.tsx

```diff
@@ -34,13 +43,29 @@ function VideoRecommendationPage() {
   }
 
   useEffect(() => {
+    if (location.search === locationSearchRef.current) return;
+    locationSearchRef.current = location.search;
```

Comment:
Could you explain why this is necessary? Is it to avoid duplicate requests to the server? Is that not redundant with the effect dependencies?

Reply:
I had to add new dependencies[[1]](https://github.com/tournesol-app/tournesol/pull/602/files#diff-216c121f61c465f9624a7b2c17723cdbaea037aa9a085d990423439036c4ff28L43-R68) to the effect hook, but we don't want to trigger a reload when these new dependencies are changed (but we still want the function to be updated).

In this case `searchParams` will change at the same time as the location so it's not a problem, but `history` may change at any time. It probably doesn't, but to be safe and to handle future changes where new dependencies may really change, I added this check to make sure we trigger a reload only when the location changes.

Reply:
> (but we still want the function to be updated)

Actually we don't really care in this case, but removing the dependencies would raise a warning in the linter.

Reply:
Ok I've just added a comment to explain this.